imblearn/under_sampling/_prototype_generation/_cluster_centroids.py

```diff
@@ -123,13 +123,8 @@ def _validate_estimator(self):
             )
         if self.estimator is None:
             self.estimator_ = KMeans(random_state=self.random_state)
-        elif isinstance(self.estimator, KMeans):
-            self.estimator_ = clone(self.estimator)
         else:
-            raise ValueError(
-                f"`estimator` has to be a KMeans clustering."
-                f" Got {type(self.estimator)} instead."
-            )
+            self.estimator_ = clone(self.estimator)
```

Comment:
This change removes the `isinstance` check for `ClusterCentroids`, similar to how [`KMeansSMOTE`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/f177b05256fa3e4e556c1a8be0283b721b1bcd86/imblearn/over_sampling/_smote/cluster.py#L148-L158) does not require such a check.